app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php

```diff
@@ -389,6 +389,65 @@ public function findLeadsToCreate($integration, $leadFields, $limit = 25, $fromD
     }
 
     /**
+     * @param $leadId
+     * @param $integration
+     * @param $integrationEntity
+     * @param $internalEntity
+     *
+     * @return int
+     */
+    public function getIntegrationEntityCount($leadId, $integration = null, $integrationEntity = null, $internalEntity = null)
+    {
+        return $this->getIntegrationEntityByLead($leadId, $integration, $integrationEntity, $internalEntity, false);
+    }
+
+    /**
+     * @param $leadId
+     * @param $integration
+     * @param $integrationEntity
+     * @param $internalEntity
+     * @param int|bool $limit
+     *
+     * @return array|int
+     */
+    public function getIntegrationEntityByLead($leadId, $integration = null, $integrationEntity = null, $internalEntity = null, $limit = 100)
+    {
+        $q = $this->_em->getConnection()->createQueryBuilder()
+            ->from(MAUTIC_TABLE_PREFIX.'integration_entity', 'i');
+
+        if (false === $limit) {
+            $q->select('count(*) as total');
+        } else {
+            $q->select('i.integration, i.integration_entity, i.internal_entity, i.integration_entity_id, i.date_added, i.last_sync_date, i.internal');
+        }
+
+        $q->where(
```

Comment:
These where statements need to be ordered to match one of the existing indexes for this table.